### ISSUE TITLE
Can I change the s2member-files directory location?

### ISSUE THREAD

**Issue by jaswrks** [websharks/s2member-kb#116]
---
##### KB Article Creation Checklist
- [x] Write initial draft for this KB Article; label this issue `draft` and either `questions` or `tutorials`
- [x] Add [required YAML configuration](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration)
- [x] Add Tags for this KB Article to the YAML config (see [YAML Keys (Explained)](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration#yaml-keys-explained))
- [x] Edit and finalize draft for publishing (remove `draft` label, add `draft-finalized` label)
- [x] Assign Issue to yourself and [create Markdown file](https://github.com/websharks/team/wiki/KB-Articles-::-Creating-the-Markdown-File) (remove `draft-finalized` label, add `pending`)
- [x] Project Lead: Review and Publish KB Article (remove `pending` label, add `published` label)
##### Additional TODOs

_Additional TODO list items go here._

---

**Markdown File:** http://wsharks.com/1Mvyl13
**Published KB Article:** http://s2member.com/kb-article/can-i-change-the-s2member-files-directory-location/

---

**Comment by bmeucci** [comment]
There's no relative path in the example.

Is the path assumed to be below the main wordpress directory or do is need to use ../ to go up a level from the s2member-files path location? What if the directory is on the same level as wordpress; will it still work?

**Comment by jaswsinc** [comment]
Thanks for the feedback. This should be a full absolute path on the server, not relative. I have updated the article to help clarify this.

In other words, the directory can live anywhere you like, you just need to type the full absolute server path to that directory.

**Comment by deadparrotsoftware** [comment]
would you not also have to make sure the new directory has the same .htaccess file as the normal protected files directory??

AND - would sub-directories also be protected?

**Comment by raamdev** [comment]
@deadparrotsoftware I believe the answer to both of those questions is yes, but I'll let Jason update the article here in case there's something I'm missing. Thanks for the feedback! :-)

**Comment by jaswrks** [comment]
> would you not also have to make sure the new directory has the same .htaccess file as the normal protected files directory??

Yes, you should do that. However, s2Member will also take care of this automatically, if it can; i.e., if the directory is writable, s2Member will automatically create the `.htaccess` file in that directory once you alter the configuration via the filter mentioned in the article. Doesn't hurt to create it yourself though, or just copy it from the old `s2member-files` location.

**Comment by jaswrks** [comment]
> AND - would sub-directories also be protected?

Yes, anything and everything in the protected directory is protected via the `.htaccess` that s2Member uses there.

**Comment by adkink** [comment]
I do everything that is explained in the example but I still receive two error messages:

 1) The security-enabled files directory (http:/localhost/s2Member/downloads) does not exist. Please create this directory manually.
 2) The .htaccess protection file (http:/localhost/s2Member/downloads/.htaccess) does not exist. Please create this file manually.

But the directory and the .htaccess file already exists.
Does anyone have an idea why this can happen?

Thank you in advance for your time.

PS: I've tried it on an online server and local (localhost). Both times with the same errors.
